packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -227,13 +263,29 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
     }
   }
 
+  WidgetStateProperty<Color?> get _widgetFillColor {
+    return WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+      if (states.contains(WidgetState.disabled)) {
+        return widget.inactiveColor;
+      }
+      if (states.contains(WidgetState.selected)) {
+        return widget.activeColor;
+      }
+      return null;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
-    final Color effectiveActiveColor = widget.activeColor
-      ?? CupertinoColors.activeBlue;
-    final Color? inactiveColor = widget.inactiveColor;
-    final Color effectiveInactiveColor = inactiveColor
-      ?? CupertinoColors.inactiveGray;
+     // Colors need to be resolved in selected and non selected states separately
+    final Set<WidgetState> activeStates = states.union({WidgetState.selected});
+    final Set<WidgetState> inactiveStates = states.difference({WidgetState.selected});
```

Comment:
It looks like this change is causing a problem with [Linux Analyze](https://github.com/flutter/flutter/pull/151761/checks?check_run_id=27626415549), since the sets need a `<WidgetState>` type annotation.

The most readable solution is probably to switch back to how it was before:

```suggestion
    final Set<WidgetState> activeStates = states..add(WidgetState.selected);
    final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
```